guvnor-ala/guvnor-ala-wildfly-provider/src/main/java/org/guvnor/ala/wildfly/access/WildflyClient.java

```diff
@@ -0,0 +1,296 @@
+        try {
+            operation.writeBase64( bout );
+        } catch ( IOException ex ) {
+            getLogger( WildflyClient.class.getName() ).log( SEVERE, null, ex );
+        }
+
+        // the multipart
+        MultipartEntityBuilder builder = create();
+        builder.setMode( BROWSER_COMPATIBLE );
+        builder.addPart( "uploadFormElement", fileBody );
+        builder.addPart( "operation", new ByteArrayBody( bout.toByteArray(), create( "application/dmr-encoded" ), "blob" ) );
+        HttpEntity entity = builder.build();
+
+        post.setEntity( entity );
+
+        try {
+            HttpResponse response = httpclient.execute( post );
+
+            return response.getStatusLine().getStatusCode();
+        } catch ( IOException ex ) {
+            ex.printStackTrace();
+            getLogger( WildflyClient.class.getName() ).log( SEVERE, null, ex );
+        }
+        return -1;
+    }
+
+    public int undeploy( String deploymentName ) {
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+                new AuthScope( host, managementPort ),
+                new UsernamePasswordCredentials( user, password ) );
+
+        CloseableHttpClient httpclient = custom()
+                .setDefaultCredentialsProvider( credsProvider )
+                .build();
+
+        final HttpPost post = new HttpPost( "http://" + host + ":" + managementPort + "/management" );
+
+        post.addHeader( "X-Management-Client-Name", "GUVNOR-ALA" );
+
+        // the DMR operation
+        ModelNode operation = new ModelNode();
+        operation.get( "operation" ).set( "remove" );
+        operation.get( "address" ).add( "deployment", deploymentName );
+
+        try {
+            post.setEntity( new StringEntity( operation.toJSONString( true ), APPLICATION_JSON ) );
+        } catch ( Exception e ) {
+            e.printStackTrace();
+        }
+
+        try {
+            HttpResponse response = httpclient.execute( post );
+            return response.getStatusLine().getStatusCode();
+        } catch ( IOException ex ) {
+            ex.printStackTrace();
+            getLogger( WildflyClient.class.getName() ).log( SEVERE, null, ex );
+        }
+        return -1;
+    }
+
+    public void close() {
+
+    }
+
+    public int start( String deploymentName ) {
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+                new AuthScope( host, managementPort ),
+                new UsernamePasswordCredentials( user, password ) );
+
+        CloseableHttpClient httpclient = custom()
+                .setDefaultCredentialsProvider( credsProvider )
+                .build();
+
+        final HttpPost post = new HttpPost( "http://" + host + ":" + managementPort + "/management" );
+
+        post.addHeader( "X-Management-Client-Name", "GUVNOR-ALA" );
```

Comment:
Out of interest why does ```undeploy(...)```, ```start(...)``` and ```stop(...)``` use ```GUVNOR-ALA``` whereas ```deploy(...)``` uses ```HAL```. Aren't all four operations part of JBoss DMR's API?